editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/SearchResultCard/DataDisplayMap.tsx

```diff
@@ -0,0 +1,116 @@
+import { ComponentType, IndexedNode } from "@opensystemslab/planx-core/types";
+import { Calculate } from "@planx/components/Calculate/model";
+import { FileUploadAndLabel } from "@planx/components/FileUploadAndLabel/model";
+import { List } from "@planx/components/List/model";
+import { SearchResult } from "hooks/useSearch";
+import { capitalize, get } from "lodash";
+import { SLUGS } from "pages/FlowEditor/data/types";
+import { useStore } from "pages/FlowEditor/lib/store";
+
+interface DataDisplayValues {
+  displayKey: string;
+  getIconKey: (result: SearchResult<IndexedNode>) => ComponentType;
+  getTitle: (result: SearchResult<IndexedNode>) => string;
+  getHeadline: (result: SearchResult<IndexedNode>) => string;
+  getComponentType: (result: SearchResult<IndexedNode>) => string;
+}
+
+/**
+ * Map of data keys to their associated display values
+ * Uses Partial<DataDisplayValues> as not all values are unique, we later apply defaults
+ */
+type DataKeyMap = Record<string, Partial<DataDisplayValues>>;
+
+/**
+ * Map of ComponentTypes to their associated data keys
+ */
+type ComponentMap = Record<ComponentType, DataKeyMap>;
+
+/**
+ * Map of ComponentTypes which need specific overrides in order to display their data values
+ */
+const DISPLAY_DATA: Partial<ComponentMap> = {
+  // Answers are mapped to their parent questions
+  [ComponentType.Answer]: {
+    default: {
+      getIconKey: () => ComponentType.Question,
+      displayKey: "Option (data)",
+      getTitle: ({ item }) => {
+        const parentNode = useStore.getState().flow[item.parentId];
+        return parentNode.data.text;
+      },
+      getHeadline: ({ item, key }) => get(item, key)?.toString(),
+    },
+  },
+  // FileUploadAndLabel has data values nested in FileTypes
+  [ComponentType.FileUploadAndLabel]: {
+    default: {
+      displayKey: "File type (data)",
+      getHeadline: ({ item, refIndex }) =>
+        (item["data"] as unknown as FileUploadAndLabel)["fileTypes"][refIndex][
+          "fn"
+        ],
```

Comment:
There's a fair amount of type casting in this data structure as `IndexedNode` (and our flow as a whole) are not strongly typed. 

We can't use a type guard as the types of `IndexedNode["data"]` and various components have no overlap - we hit the `"A type predicate's type must be assignable to its parameter's type"` error.

This means that casting is unfortunately necessary, but it's better than using `any` or just optional chaining the whole way down.